File: implementations/csharp/src/Loom.Core/Models/RagChunk.cs
using System.Collections.Generic;

namespace Loom.Core.Models
{
    /// <summary>
    /// Single chunk of the list
    /// </summary>
    public class RagChunk
    {
        public string SourceId { get; set; }
        public double Score { get; set; } // Semantic relevance (0.0 - 1.0)
        public string Text { get; set; }
        public Dictionary<string, object> Metadata { get; set; }
    }
}

File: implementations/vb/Loom.Core/Models/RagChunk.vb
Namespace Models
    ''' <summary>
    ''' Single chunk of the list
    ''' </summary>
    Public Class RagChunk
        Public Property SourceId As String
        Public Property Score As Double ' Semantic relevance (0.0 - 1.0)
        Public Property Text As String
        Public Property Metadata As New Dictionary(Of String, Object)
    End Class
End Namespace